ddtrace/tracer/tracer.go

```diff
@@ -143,24 +169,22 @@ func newTracer(opts ...StartOption) *tracer {
 		prioritySampling: newPrioritySampler(),
 		pid:              strconv.Itoa(os.Getpid()),
 	}
+	t.config.statsd.Incr("datadog.tracer.started", nil, 1)
 	if c.runtimeMetrics {
-		statsd, err := statsd.NewBuffered(t.config.dogstatsdAddr, 40)
-		if err != nil {
-			log.Warn("Runtime metrics disabled: %v", err)
-		} else {
-			log.Debug("Runtime metrics enabled.")
-			go t.reportMetrics(statsd, defaultMetricsReportInterval)
-		}
+		log.Debug("Runtime metrics enabled.")
+		t.wg.Add(1)
+		go t.reportRuntimeMetrics(defaultMetricsReportInterval)
 	}
-
+	t.wg.Add(2)
```

Comment:
I feel that it can easily lead to errors when adding to a waitgroup is not closely coupled with the thing being added and the the completion of it. Even if uglier, I feel it's more safe to do it like this (it's what we've been doing elsewhere too, like in our statsd PR recently):
```
t.wg.Add(1)
go func() {
    defer t.wg.Done()
    t.worker()
}()

t.wg.Add(1)
go func() {
    defer wg.Done()
    t.reportHealthMetrics()
}()
```
...and the same for runtime metrics above (currently line 176).

Otherwise, it's odd seeing those `defer wg.Done()`'s and they can seem out of place. Also, it's very easy to commit an error. This way they are tightly coupled and obvious.

Reply:
This is much better.